tools/wake-unit/term.cpp

```diff
@@ -384,3 +387,250 @@ TEST(term_dumb_gcc_smoke_test) {
       "\033[01m\033[Ktools/wake-unit/unit.h:146:95:\033[m\033[K \033[01;31m\033[Kerror:",
       "tools/wake-unit/unit.h:146:95: error:");
 }
+
+TEST(term_git_smoke_test) {
+  CONVERTS_TO("\033[?1h\033=\r2023-02-08 12:31:48 -0600\033[m\r\n\r\033[K\033[?1l\033>",
+              "2023-02-08 12:31:48 -0600\033(B\033[m\n");
+}
+
+TEST(term_dumb_git_smoke_test) {
+  CONVERTS_TO_DUMB("\033[?1h\033=\r2023-02-08 12:31:48 -0600\033[m\r\n\r\033[K\033[?1l\033>",
+                   "2023-02-08 12:31:48 -0600\n");
+}
+
+TEST(term_unicode_smoke_test) {
+  // Unicode should be ignored even if there's an escape sequence
+  // or other special character embeded inside
+  // CONVERTS_TO("\xC0\r", "\xC0\r");
+  CONVERTS_TO("\xC0\033", "\xC0\033");
+
+  CONVERTS_TO("\xE0\x80\r", "\xE0\x80\r");
+  CONVERTS_TO("\xE0\x80\033", "\xE0\x80\033");
+
+  CONVERTS_TO("\xF0\x80\x80\r", "\xF0\x80\x80\r");
+  CONVERTS_TO("\xF0\x80\x80\033", "\xF0\x80\x80\033");
+
+  // However if we go just one more we'll fall out of the unicode states
+  // and special characters should be removed
+  CONVERTS_TO("\xF0\x80\x80\x80\r", "\xF0\x80\x80\x80");
+}
+
+TEST(term_dumb_unicode_smoke_test) {
+  // Unicode should be ignored even if there's an escape sequence
+  // or other special character embeded inside
+  CONVERTS_TO_DUMB("\xC0\r", "\xC0\r");
+  CONVERTS_TO_DUMB("\xC0\033", "\xC0\033");
+
+  CONVERTS_TO_DUMB("\xE0\x80\r", "\xE0\x80\r");
+  CONVERTS_TO_DUMB("\xE0\x80\033", "\xE0\x80\033");
+
+  CONVERTS_TO_DUMB("\xF0\x80\x80\r", "\xF0\x80\x80\r");
+  CONVERTS_TO_DUMB("\xF0\x80\x80\033", "\xF0\x80\x80\033");
+
+  // However if we go just one more we'll fall out of the unicode states
+  // and special characters should be removed
+  CONVERTS_TO_DUMB("\xF0\xFF\xFF\xFF\r", "\xF0\xFF\xFF\xFF");
+}
```

Comment:
These are basically testing that unicode works the way that we expect it to. Namely that it interprets the leading UTF-8 character as a marker to get the length of the character.